src/AutoRest.CSharp/Mgmt/Output/Resource.cs

```diff
@@ -127,8 +127,8 @@ protected internal Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operat
             yield return new FieldDeclaration(FieldModifiers, ResourceData.Type, DataFieldName);
         }
 
-        public Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operation>> allOperations, string resourceName, ResourceTypeSegment resourceType, ResourceData resourceData, BuildContext<MgmtOutputLibrary> context)
-            : this(allOperations, resourceName, resourceType, resourceData, context, ResourcePosition)
+        public Resource(IReadOnlyDictionary<OperationSet, IEnumerable<Operation>> allOperations, string resourceDataSchemaName, ResourceTypeSegment resourceType, ResourceData resourceData, BuildContext<MgmtOutputLibrary> context)
```

Comment:
Could you please add a `TODO` here so that we could track the refactor of making a 1-1 map between OperationSet and Resource?
I will make the refactor in another PR once yours merges